test/e2e/network_policy.go

```diff
@@ -0,0 +1,509 @@
+		Expect(err).NotTo(HaveOccurred(), "checking Client-A could communicate with server.")
+
+		framework.Logf("Waiting for client-b to complete.")
+		err = framework.WaitForPodSuccessInNamespace(f.Client, podClientB.Name, ns.Name)
+		Expect(err).To(HaveOccurred(), "checking Client-B could not communicate with server")
+	})
+
+	It("should enforce policy based on NamespaceSelector [Feature:NetworkPolicy]", func() {
+		nsA := f.Namespace
+		nsBName := f.BaseName + "-b"
+		// The CreateNamespace helper uses the input name as a Name Generator, so the namespace itself
+		// will have a different name than what we are setting as the value of ns-name.
+		// This is fine as long as we don't try to match the label as nsB.Name in our policy.
+		nsB, err := f.CreateNamespace(nsBName, map[string]string{
+			"ns-name": nsBName,
+		})
+		Expect(err).NotTo(HaveOccurred())
+		setNamespaceIsolation(f, nsA, "DefaultDeny")
+		setNamespaceIsolation(f, nsB, "DefaultDeny")
+
+		// Create Server with Service in NS-B
+		By("Creating a webserver tied to a service.")
+		podServer, service := createServerPod(f, nsA, serverName, []int{listeningPort1})
+		defer func() {
+			By("Cleaning up the server.")
+			if err := f.Client.Pods(nsA.Name).Delete(podServer.Name, nil); err != nil {
+				framework.Failf("unable to delete pod %v: %v", podServer.Name, err)
+			}
+		}()
+		defer func() {
+			By("Cleaning up the server's service.")
+			if err := f.Client.Services(nsA.Name).Delete(service.Name); err != nil {
+				framework.Failf("unable to delete svc %v: %v", service.Name, err)
+			}
+		}()
+		framework.Logf("Waiting for server to come up.")
+		err = framework.WaitForPodRunningInNamespace(f.Client, podServer)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Create Policy for that service that allows traffic only via namespace B
+		By("Creating a network policy for the server which allows traffic from namespace-b.")
+		policy, err := f.Client.NetworkPolicies(nsA.Name).Create(&extensions.NetworkPolicy{
+			ObjectMeta: api.ObjectMeta{
+				Name: "test-policy",
+			},
+			Spec: extensions.NetworkPolicySpec{
+				// Apply to server
+				PodSelector: unversioned.LabelSelector{
+					MatchLabels: map[string]string{
+						"pod-name": podServer.Name,
+					},
+				},
+				// Allow traffic only from NS-B
+				Ingress: []extensions.NetworkPolicyIngressRule{{
+					From: []extensions.NetworkPolicyPeer{{
+						NamespaceSelector: &unversioned.LabelSelector{
+							MatchLabels: map[string]string{
+								"ns-name": nsBName,
+							},
+						},
+					}},
+				}},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			By("Cleaning up the policy")
+			if err := f.Client.NetworkPolicies(nsA.Name).Delete(policy.Name, nil); err != nil {
+				framework.Failf("unable to delete policy %v: %v", policy.Name, err)
+			}
+		}()
+
+		// Create a pod with name 'client-a', which should be able to communicate with server.
```

Comment:
No, Casey's original comment was correct; line 347 says:

    // Create a pod with name 'client-a', which should be able to communicate with server.

and 348 says:

     By("Creating client-a in ns-a which should *not* be able to contact the server.")

(or maybe the comment got moved around after new commits were pushed...)